Remove incorrect validation of app service sticky setting names

jennings opened the pull request:
This validation is too strict. The Azure Portal allows the following name for both app settings and connection strings:

    A test: !@#$%^&*()_+-=' ";/?

I can't find a reference for what the valid character set would be, and the names in the app_settings attribute aren't validated at all, so we may as well remove this validation.

Fixes #16612

![image](https://user-images.githubusercontent.com/24511/173157740-4b21fa30-979e-4dc3-ab5a-b8e7683b708f.png)

jennings commented:
@jackofallops I would love to get this into an upcoming release because it's blocking my use of `sticky_settings` entirely. Is there any research I can to do make this easier to review? Or, let me know if I should change something and I can work on it.

jackofallops commented on internal/services/appservice/helpers/shared_schema.go:
```diff
@@ -1554,7 +1553,6 @@ func StickySettingsSchema() *pluginsdk.Schema {
 					Optional: true,
 					Elem: &pluginsdk.Schema{
 						Type:         pluginsdk.TypeString,
-						ValidateFunc: appserviceValidate.AppSettingName,
```
Unfortunately, we can't remove validation entirely, since an empty name is invalid, so we'd need to at least use:
```
							ValidateFunc: validation.StringIsNotEmpty,
```

jackofallops commented on internal/services/appservice/helpers/shared_schema.go:
```diff
@@ -1568,7 +1566,6 @@ func StickySettingsSchema() *pluginsdk.Schema {
 					Optional: true,
 					Elem: &pluginsdk.Schema{
 						Type:         pluginsdk.TypeString,
-						ValidateFunc: appserviceValidate.AppSettingName,
```
as above.

jackofallops reviewed (changes requested):
HI @jennings - Apologies for the delay in review, I've been away for a few days. 

Thanks for the catch here, the information I developed this against suggested this was the correct validation, but I should have remembered your point about setting hierarchy, my bad!

Just two minor changes needed, since we can't remove validation entirely here.

Can you add test(s) for these names that cover special characters?  Since this is something that can potentially change over time, we'd need to ensure that's covered.  Once those changes and tests are in I'll run them on our acceptance suite and we'll get this merged.

Thanks again!

jennings commented:
@jackofallops I added `validation.StringIsNotEmpty` and added my pathological setting name to the four acceptance tests I found that are testing `sticky_settings` on `azurerm_windows_web_app` and `azurerm_windows_function_app`. I hope this is what you intended, if not I can try again. Thanks for your help!

jackofallops commented:
Tests looking fine:
![image](https://user-images.githubusercontent.com/11830746/175289133-b74fe201-54bb-4966-8d55-3adadb040cc1.png)

Failure here was a transient issue:
![image](https://user-images.githubusercontent.com/11830746/175466214-66a9ce4c-529f-474c-b21d-ca1465c21f99.png)
